src/models/iceflow/SIA2D/SIA2D_utils.jl

```diff
@@ -266,4 +266,40 @@ function surface_V(H::Matrix{F}, simulation::SIM) where {F <: AbstractFloat, SIM
     Vy = - D .* avg_x(dSdy)
 
     return Vx, Vy    
-end
\ No newline at end of file
+end
+
+function H_from_V(V::Matrix{<:Real}, simulation::SIM) where {SIM <: Simulation}
+    params::Sleipnir.Parameters = simulation.parameters
+    
+    iceflow_model = simulation.model.iceflow
+    glacier::Sleipnir.Glacier2D = simulation.glaciers[iceflow_model.glacier_idx[]]
+    B = glacier.B
+    Δx = glacier.Δx
+    Δy = glacier.Δy
+    A = iceflow_model.A
+    n = iceflow_model.n
+    C = iceflow_model.C
+    ρ = params.physical.ρ
+    g = params.physical.g
+    H₀ = glacier.H₀
+
+    # Update glacier surface altimetry
+    S = glacier.S  
```

Comment:
This means that `H` will always be computed with the initial surface altimetry. Shouldn't `S` be taken from `simulation` instead of `glacier`? 

For context, `glacier` stores the initial conditions of the glacier based on observations. `simulation.model.iceflow`, which in your case is a 2D SIA model, contains the current solution. If you intend to use this in a forward model, you should take `S` from `simulation.model.iceflow`

Reply:
Yes I think you are right. I will check this to be certain and change it.